docs/cli.md

```diff
@@ -0,0 +1,322 @@
+{#cli}
+
+# Command-line interface
+
+Running
+`jaq`\ \[_OPTION_\]...\ \[_FILTER_\]\ \[_FILE_\]...
+performs the following steps:
+
+- Parse _FILTER_ as jq program; see [jq language](#jq-lang)
+- For each _FILE_:
+
+    - Parse _FILE_ to a stream of values
+    - For each input value in the file:
+
+        - Run _FILTER_ on the input value and print its output values
+
+For example, `jaq '.name?' persons.json`
+parses the filter `.name?`, then
+reads all values in `persons.json` one-by-one.
+It then executes the filter `.name?` on each of the values and
+prints the outputs of the filter as JSON.
+
+There are a few rules:
+
+- If no _FILTER_ is given, jaq uses `.` (the [identity] filter) as filter.
+- If no _FILE_ is given, jaq reads from standard input.
+- jaq determines the format to parse a _FILE_ as follows:
+
+    - If [`--from`](#--from) _FORMAT_ is used, jaq uses that format.
+    - Otherwise, if _FILE_ has a file extension known by jaq, such as
+      `.json`, `.yaml`, `.cbor`, `.toml`, `.xml`,
+      jaq uses the corresponding format.
+    - Otherwise, jaq assumes JSON.
+- If an error is encountered at any point, jaq stops.
+
+When passing filters directly as _FILTER_ argument on the command-line,
+care has to be taken to properly escape the filter.
+How to do this depends from platform to platform, but on Unixoid systems,
+surrounding the filter with single quotes (`'`) and
+replacing occurrences of `'` in filters by `'\''` suffices.
+For example, to run the filter `"'"` that
+produces a string containing a single quote, you can use
+`jaq -n '"'\''"'`.
+
+Running filters that start with the [negation operator](#negation),
+such as `jaq '-1'`, fails because `-` is interpreted as
+start of a command-line switch rather than negation.
+You can remedy this by using
+`jaq -- '-1'` instead, or by surrounding the filter in parentheses, i.e.
+`jaq '(-1)'`.
+
+
+## Input options
+
+{#--from}
+
+### `--from` _FORMAT_
+
+Interpret all input files as _FORMAT_.
+For example,
+`jaq --from yaml . myfile.yml`
+parses `myfile.yml` as YAML.
+Possible values of _FORMAT_ include:
+`raw`, `json`, `yaml`, `cbor`, `toml`, `xml`.
+
+jaq automatically chooses the corresponding input format for
+files with the extensions
+`.json`, `.yaml`, `.cbor`, `.toml`, `.xml`, `.xhtml`.
+That means that
+`jaq --from cbor . myfile.cbor` is equivalent to
+`jaq . myfile.cbor`.
```

Comment:
Is there a way to read a mix of formats? `jaq . a.xml b.cbor` and how does `--from` affect format based on extension? ex `jaq --from cbor . im-cbor im.xml`